docs/drvfc.rst

```diff
@@ -0,0 +1,152 @@
+    </devices>
+  </domain>
+
+In the above example we specified the domain type to be `firecracker`,
+we specified the name of the domain, together with the memory and vcpus allocated.
+In the `<os>` tag we specified the source file for the kernel binary, the kernel command line.
+The only architecture supported for now is `x86_64` with the `hvm` type.
+The `<root>` tag specifies which of the following disk devices is used as the rootfs by matching the
+dev mount point attribute in the `<target>` tag.
+
+For the devices section:
+The only disk device is actually the rootfs, as it corresponds to the <root>
+tag from the os definition
+There is one serial device of type `pty` with target port `0`, however the target port is
+not relevant depending on the version of firecracker.
+Below there are 2 net interfaces defined of type ethernet, each connected
+to already existing tapa devices
+
+XML structure:
+===============
+
+``<domain>``
+   The domain type should be ``firecracker``
+
+``<name>``
+   The name tag specifies the name of the vm
+
+``<memory>``
+   The memory tag specifies the amount of RAM given to the VM.
+
+``<vcpu>``
+   The number of vcpu threads allocated for the vm
+
+``<os>``
+   Includes information about os configuration
+   Currently only ``x86_64`` architecture is supported, having ``hvm`` type
+
+   ``<kernel>``
+      Path to the kernel binary the vm will run on.
+
+   ``<cmdline>``
+      The command line arguments the vm will boot with
+
+   ``<root>``
+      The value of this tag tells the driver which one of the disk devices
+      is the rootfs one. It matches this value with the attribute `dev` on the ``<target>`` tag of
+      the device
+
+``<devices>``
+   Devices supported include disk devices and maximum one serial device.
+
+   **Disk devices:**
+
+   Currently for disk devices, only one disk device is supported, the rootfs, one example below:
+
+   ::
+
+      <disk type='file' device='disk'>
+         <source file='/home/ec2-user/LV/myownfiles/isos/hello-rootfs.ext4'/>
+         <target dev='vda' bus='virtio' />
+         <readonly/>
+      <disk>
+
+
+   ``<source>``
+      Mandatory tag that specifies the path to the backing file for the disk device
+
+   ``<target>``
+      Mandatory tag that.
+      Specifies the drive id under the ``dev`` attribute.
+      Currently only virtio devices are supported, so this tag should be present: ``bus='virtio'``
+
+   ``<readonly/>``
+      Optional tag that specifies if the disk device should be mounted in read-only mode
+
+   **Serial devices:**
+
+   Firecracker supports only one serial console device. If you wish to use make the serial device available
+   you need to specify it using a ``<serial>`` device. An example below:
+
+   ::
+
+      <serial type='pty'>
+         <target port='0'/>
+      </serial>
+
+
+   ``<target>``
+      Optional tag that specifies the port to be useed for the serial. However this property is currently ignored in firecracker `v0.*`
```

Comment:
```suggestion
      Optional tag that specifies the port to be used for the serial. However this property is currently ignored in firecracker `v0.*`
```

Reply:
add details at the top about that Firecracker version this was tested on and is supported on